ynr/apps/candidates/mastodon_api.py

```diff
@@ -0,0 +1,46 @@
+import requests
+from django.conf import settings
+from django.core.cache import cache
+
+
+class MastodonAPITokenMissing(Exception):
+    pass
+
+
+def get_mastodon_user_id(mastodon_screen_name):
+    cache_key = "mastodon-screen-name:{}".format(mastodon_screen_name)
+    cached_result = cache.get(cache_key)
+    if cached_result:
+        return cached_result
+    token = settings.MASTODON_APP_ONLY_BEARER_TOKEN
+    if not token:
+        raise MastodonAPITokenMissing()
+    headers = {"Authorization": "Bearer {token}".format(token=token)}
+
+    r = requests.get(
+        "https://mstdn.social/api/v1/accounts/",
```

Comment:
We're going to want to take the domain name from the user account, as we'll need to ask their server about them. 

This might get tricky as some servers aren't going to have the API turned on. In that case I think we just need to trust that it's a valid account, but I'm not sure how we tell the difference between an invalid account and a non-existent API.

My server returns a 404 for a user name that doesn't exist:

https://mastodon.me.uk/api/v1/accounts/lookup?acct=notsymroe

But I imagine it would also return a 404 if the API wasn't enabled!